src/main/java/org/jboss/pnc/common/sql/NativeQueryBuilder.java

```diff
@@ -0,0 +1,133 @@
+package org.jboss.pnc.common.sql;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class NativeQueryBuilder {
+
+    private final List<String> selects;
+    private final List<String> froms;
+    private final List<String> joins;
+    private final List<String> wheres;
+    private final List<String> orderBys;
+    private Integer limit;
+    private Integer offset;
+
+    private NativeQueryBuilder() {
+        selects = new ArrayList<>();
+        froms = new ArrayList<>();
+        joins = new ArrayList<>();
+        wheres = new ArrayList<>();
+        orderBys = new ArrayList<>();
+    }
+
+    public NativeQueryBuilder select(String field) {
+        selects.add(field);
+        return this;
+    }
+
+    public NativeQueryBuilder select(String table, String field) {
+        selects.add((table == null || table.isEmpty() ? "" : table + '.') + field);
+        return this;
+    }
+
+    public NativeQueryBuilder select(String table, String field, String alias) {
+        selects.add(
+                (table == null || table.isEmpty() ? "" : table + '.') + field
+                        + (alias == null || alias.isEmpty() ? "" : " AS " + alias));
```

Comment:
I would extract `(alias == null || alias.isEmpty() ? "" : " AS " + alias)` into a method as it's commonly used